RetroAchievements/RAWeb
# refactor: implement almost all forum table TODOs

This PR adds 4 migrations: one for each of the forum tables.

Before going down the path of migrating forum stuff to React, it would probably make a lot of sense to get these TODOs out of the way.

* ForumCategory -> forum_categories
  * ID -> Id
  * Name -> title
  * Description -> description
  * DisplayOrder -> order_column
  * Created -> created_at
  * Updated -> updated_at

* Forum -> forums
  * ID -> id
  * CategoryID -> forum_category_id
  * Title -> title
  * Description -> description
  * LatestCommentID -> latest_comment_id
  * DisplayOrder -> order_column
  * Created -> created_at
  * Updated -> updated_at

* ForumTopic -> forum_topics
  * ID -> id
  * ForumID -> forum_id
  * Title -> title
  * DateCreated -> created_at
  * Updated -> updated_at
  * LatestCommentID -> latest_comment_id
  * RequiredPermissions -> required_permissions

* ForumTopicComment -> forum_topic_comments
  * ID -> id
  * ForumTopicID -> forum_topic_id
  * Payload -> body
  * DateCreated -> created_at
  * DateModified -> updated_at
  * Authorised -> is_authorized

There should be no functional changes as part of this PR.

Behaviors I've verified:
* Create a game's official forum topic
* Manage games in Filament
* View user's recent forum posts
* Visit recent posts page
* Open topics
* Create topics
* Edit topics
* View and paginate through topics
* Observe email notifications on topic replies
* Unsubscribe/subscribe on topics
* Navigate through forum categories
* Paginate through forum categories
* Authorize posts
